api/filter/http_connection_manager.proto

```diff
@@ -0,0 +1,250 @@
+  // an additional random jitter to the delay that is between zero and
+  // refresh_delay_ms milliseconds. Thus the longest possible refresh delay is 2
+  // * refresh_delay_ms. Default value is 30000ms (30 seconds).
+  google.protobuf.Duration refresh_delay = 3;
+}
+
+// Filters on HTTP response/status code.
+message StatusCodeFilter {
+  // Comparison operator.
+  enum Op {
+    EQ = 0;
+    GE = 1;
+  }
+  Op op = 1;
+
+  // Default value to compare against if runtime value is not available.
+  google.protobuf.UInt32Value value = 2;
+
+  // Runtime key to get value for comparision. This value is used if defined.
+  string runtime_key = 3;
+}
+
+// Filters on total request duration.
+message DurationFilter {
+  // Comparison operator.
+  enum Op {
+    EQ = 0;
+    GE = 1;
+  }
+  Op op = 1;
+
+  // Default value to compare against if runtime value is not available.
+  google.protobuf.Duration value = 2;
+
+  // Runtime key to get value for comparision. This value is used if defined.
+  string runtime_key = 3;
+}
+
+// Filters for requests that are not health check requests. A health check
+// request is marked by the health check filter.
+message NotHealthCheckFilter {
+}
+
+// Filters for requests that are traceable. See the tracing overview for more
+// information on how a request becomes traceable.
+message TraceableFilter {
+}
+
+// Filters for random sampling of requests. Sampling pivots on the header
+// x-request-id being present. If x-request-id is present, the filter will
+// consistently sample across multiple hosts based on the runtime key value and
+// the value extracted from x-request-id. If it is missing, the filter will
+// randomly sample based on the runtime key value.
+message RuntimeFilter {
+  // Runtime key to get the percentage of requests to be sampled. This runtime
+  // control is specified in the range 0-100 and defaults to 0.
+  string key = 1;
+}
+
+// Performs a logical “and” operation on the result of each filter in filters.
+// Filters are evaluated sequentially and if one of them returns false, the
+// filter returns false immediately.
+message AndFilter {
+  repeated AccessLogFilter filters = 1;
+}
+
+// Performs a logical “or” operation on the result of each individual filter.
+// Filters are evaluated sequentially and if one of them returns true, the
+// filter returns true immediately.
+message OrFilter {
+  repeated AccessLogFilter filters = 2;
+}
+
+message AccessLogFilter {
+  oneof filter_specifier {
+    StatusCodeFilter status_code = 1;
+    DurationFilter duration = 2;
+    NotHealthCheckFilter not_health_check = 3;
+    TraceableFilter traceable = 4;
+    RuntimeFilter runtime = 5;
+    AndFilter and = 6;
+    OrFilter or = 7;
+  }
+}
+
+message AccessLog {
+  // Path the access log is written to.
+  // TODO(dpn): Should this be a one_of and allow a remote log gRPC endpoint
+  // in conjunction with the proto access log format?
+  string path = 1;
```

Comment:
Yes, I think this is the right approach.  It should be one_of a path string, or a gRPC endpoint string.